src/formElementPlugins/ApplicationViewWrapper.tsx

```diff
@@ -103,20 +92,36 @@ const ApplicationViewWrapper: React.FC<ApplicationViewWrapperProps> = (props) =>
       type: 'setElementTimestamp',
       timestampType: 'elementLostFocusTimestamp',
     })
-    const validationResult: ValidationState = await onUpdate(jsonValue.text)
-    if (jsonValue.text !== undefined)
+
+    if (!jsonValue.customValidation) {
+      // Validate and Save response -- generic
+      const validationResult: ValidationState = await onUpdate(jsonValue.text)
+      if (jsonValue.text !== undefined)
+        await responseMutation({
+          variables: {
+            id: currentResponse?.id as number,
+            value: jsonValue,
+            isValid: validationResult.isValid,
+          },
+        })
+      if (jsonValue.text === allResponses[code]?.text) {
+        setApplicationState({
+          type: 'setElementTimestamp',
+          timestampType: 'elementsStateUpdatedTimestamp',
+        })
+      }
+    } else {
+      // Save response for plugins with internal validation
+      const { isValid, validationMessage } = jsonValue.customValidation
+      setValidationState({ isValid, validationMessage })
+      delete jsonValue.customValidation // Don't want to save this field
       await responseMutation({
         variables: {
           id: currentResponse?.id as number,
           value: jsonValue,
-          isValid: validationResult.isValid,
+          isValid,
         },
       })
-    if (jsonValue.text == allResponses[code]?.text) {
-      setApplicationState({
-        type: 'setElementTimestamp',
-        timestampType: 'elementsStateUpdatedTimestamp',
-      })
```

Comment:
I'm trying to understand why this code is on the `IF` statement (when it's a generic validation) but not in here?

Reply:
This was from an older issue PR #229. Normally the `elementsStateUpdatedTimestamp` is set when `elementsState` is updated, but if the value hasn't changed it needs to be updated here.

And since the `elementsState` isn't updated (since it's external "value" doesn't change really) then I guess it would make sense to include this in the "internal validation" block as well. Thanks for bringing it to my attention.

Done in 6734ce6